app/controllers/admin/delivery_partners_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class Admin::DeliveryPartnersController < Admin::BaseController
+  skip_after_action :verify_authorized
+  skip_after_action :verify_policy_scoped
+
+  def choose_lead_providers
+    authorize DeliveryPartner, :create?
+
+    new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    if session[:delivery_partner_form].nil?
+      session[:delivery_partner_form] = { name: new_supplier_form.name }
+    else
+      session[:delivery_partner_form].merge!({ name: new_supplier_form.name })
+    end
+
+    @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+  end
+
+  def receive_lead_providers
+    authorize DeliveryPartner, :create?
+
+    @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+    @delivery_partner_form.populate_provider_relationships(params)
+
+    unless @delivery_partner_form.valid?
+      render :choose_lead_providers and return
+    end
+
+    session[:delivery_partner_form].merge!(
+      {
+        lead_providers: @delivery_partner_form.lead_providers,
+        provider_relationships: @delivery_partner_form.provider_relationships,
+      },
+    )
+    redirect_to admin_new_delivery_partner_review_path
+  end
+
+  def review_delivery_partner
+    authorize DeliveryPartner, :create?
+    @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+  end
+
+  def create_delivery_partner
+    authorize DeliveryPartner, :create?
+
+    delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+    delivery_partner = delivery_partner_form.save!
+
+    redirect_to admin_new_delivery_partner_success_path(delivery_partner: delivery_partner)
+  rescue ActiveRecord::RecordInvalid
+    render :'errors/internal_server_error'
+  end
+
+  def delivery_partner_success
+    authorize DeliveryPartner, :create?
+    session.delete(:new_supplier_form)
+    session.delete(:lead_provider_form)
+    session.delete(:delivery_partner_form)
+
+    @delivery_partner = DeliveryPartner.find(params[:delivery_partner])
+  end
+
+  def show_users
```

Comment:
We could add a before_action for show_lps etc Probably more railsy than explicitly calling the show method in each.